### ISSUE TITLE
Can we specify `swift_version` in podspec?

### ISSUE THREAD

**Issue by coderguide** [groue/GRDB.swift#410]
Hello @groue 
Thanks for the handy library, I'm using it for a long time without any issue. Just recently I wanted to test an older version of GRDB, to see the differences between GRDB 2 and 3 while migrating, but I had a little trouble.  

Here is how it goes:

### What did you do?

I am using Xcode 10, and my project target has `SWIFT_VERSION = 4.2` in its build settings, and it uses new features from swift `4.2`. 

I added `pod 'GRDB.swift','2.10.0'` to my `Podfile` to deliberately fetch an older version of GRDB and performed `pod install`. 

### What did you expect to happen?

When I checked the build settings for `GRDB.swift` target (from Pods project), I expected to see `SWIFT_VERSION` being set as `4`, since GRDB `2.10.0` is only compatible with `Swift 4`.

### What happened instead?

Instead, the `SWIFT_VERSION` for `GRDB.swift` target was set as `4.2`.  
Cocoapods assigned its swift version as the same from my project target, which leaded to compilation errors, due to swift incompatibility.

### Environment

**GRDB flavor(s):** GRDB
**GRDB version:** 2.10.0
**Installation method:** CocoaPods
**Xcode version:** 10
**Swift version:** 4.2 (for main app target)
**Platform(s) running GRDB:** iOS
**macOS version running Xcode:** macOS 10.14 (18A371a)

### Possible solution

In the `GRDB.swift.podspec` file, we can explicitly specify the swift version like this:
```
  s.swift_version = '4.0'
```
So this way, when the pod is installed, Cocoapods will respect the version choice from podspec, and will always use it when it sets the `SWIFT_VERSION` for  `GRDB.swift` target.   

This modification can be applied to all podspec files, according to their swift version support.

*P.S. I did my best to check if there is already a similar issue about this. Seems like this is also related to issue #379*

**Comment by coderguide** [comment]
Oppss, just found out the issue #375. 
You already made it clear why you don't prefer this solution.
I'm closing this, sorry for the noise :)

**Comment by groue** [comment]
Hello @coderguide,

The podspec does not lock any `swift_version` because I want GRDB to be able to run with multiple Swift versions (#375, yes). On top of that, CocoaPods does not allow (yet?) to specify a *range* of supported language versions.

This explains your trouble: I understand your issue, and don't be sorry!

A possible solution is in our Podfile: instruct CocoaPods to compile GRDB with the Swift version you want:

```ruby
pod 'GRDB.swift'
...

post_install do |installer|
  installer.pods_project.targets.each do |target|
    if target.name == 'GRDB.swift'
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.0'
      end
    end
  end
end
```

Now the `4.0` version, for Xcode 10, actually means Swift 4.1. I **don't know** if GRDB 2.10 can be compiled in this mode. If it does not, you may seriously consider using the latest version of GRDB. Migrating from GRDB 2 to 3 is not very hard. It ships with many fixits that help you in the process. There even is a [migration guide](https://github.com/groue/GRDB.swift/blob/master/Documentation/GRDB2MigrationGuide.md).
```ruby
post_install do |installer|
  installer.pods_project.targets.each do |target|
      if ['AirMapSDK', 'PhoneNumberKit', 'Lock', 'RxSwift', 'RxSwiftExt', 'RxCocoa', 'RxDataSources', 'ProtocolBuffers-Swift'].include? target.name
          target.build_configurations.each do |config|
              config.build_settings['SWIFT_VERSION'] = '4.0'
          end
      end
  end
end
```

**Comment by coderguide** [comment]
Thanks for your quick response with the workaround.

Although **GRDB 2** is compiling successfully with **Swift 4** compiler from **Xcode 10**, I'm planning to switch **GRDB 3** completely to be up to date. Looks like my existing code is already compatible with GRDB 3, but I'll check the migration guide thoroughly to be sure.

As you say, there is currently no support for ranged swift versions for podspecs, but they recently scheduled it as a milestone for Cocoapods `1.7.0`: 
https://github.com/CocoaPods/CocoaPods/issues/7134

I'll keep an eye on it and let you know ;)

**Comment by groue** [comment]
Since https://github.com/groue/GRDB.swift/pull/534, merged into v4.0.1, the pod spec mentions the supported `swift_versions`. Works with CocoaPods 1.7.0.

**Comment by izooto-mobile-sdk** [comment]
I got this issues,
Please give me solution
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
